ros/angel_system_nodes/angel_system_nodes/dummy_multi_task_monitor.py

```diff
@@ -0,0 +1,360 @@
+    # This may be removed from if step progress regresses.
+    steps_skipped_cache: Set[int] = field(default_factory=set)
+    # Track the latest activity classification end time sent to the HMM
+    # Time is in floating-point seconds up to nanosecond precision.
+    latest_act_classification_end_time = None
+    # List of task steps
+    steps: List[str] = field(default_factory=list)
+
+    def __post_init__(self):
+        # Boolean vector the length of task steps
+        # that indicates which steps to report as having been completed.
+        self.steps_complete = np.zeros(self.n_steps, dtype=bool)
+        # Skipped steps are those steps at or before the current
+        # skipped (haven't happened yet).
+        self.steps_skipped = np.zeros(self.n_steps, dtype=bool)
+
+
+class DummyMultiTaskMonitor(Node):
+    """
+    ROS node that simulates a node that could monitor multiple tasks
+    simultaneously. The keyboard is used to advance between task steps.
+    """
+
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+        log = self.get_logger()
+
+        param_values = declare_and_get_parameters(
+            self,
+            [
+                (PARAM_CONFIG_FILE,),
+                (PARAM_TASK_STATE_TOPIC,),
+                (PARAM_QUERY_TASK_GRAPH_TOPIC,),
+            ],
+        )
+        self._config_file = param_values[PARAM_CONFIG_FILE]
+        self._task_state_topic = param_values[PARAM_TASK_STATE_TOPIC]
+        self._query_task_graph_topic = param_values[PARAM_QUERY_TASK_GRAPH_TOPIC]
+
+        # Load the task configurations
+        with open(self._config_file, "r") as f:
+            config = yaml.safe_load(f)
+
+        self._task_state_dict: Dict[int, TaskStateInformation] = {}
+        for t in config["tasks"]:
+            # Load the config for this task
+            with open(t["config_file"], "r") as f:
+                task_config = yaml.safe_load(f)
+
+            n_steps = task_config["steps"][-1]["id"]
+
+            task_state_info = TaskStateInformation(
+                f"{t['id']}_{task_config['title']}",
+                n_steps,
+            )
+
+            # Load the list of task steps
+            task_state_info.steps = [
+                step["description"] for step in task_config["steps"]
+            ]
+
+            self._task_state_dict[t["id"]] = task_state_info
+
+        log.info("Loaded the following tasks:")
+        for key, value in self._task_state_dict.items():
+            log.info(f"    Task {key}: {value.task_title}")
+
+        # Initialize ROS hooks
+        self._task_update_publisher = self.create_publisher(
+            TaskUpdate, self._task_state_topic, 1
+        )
+        self._task_graph_service = self.create_service(
+            QueryTaskGraph, self._query_task_graph_topic, self.query_task_graph_callback
+        )
+
+        # Store module for local use.
+        self._keyboard = keyboard
```

Comment:
Why do we have to do this instead of just calling into the module normally?

Reply:
Good question. This was left over from what I copied from `task_monitor_v2`, which had the optional keyboard advance feature and a delayed import of `keyboard`. Since this module only supports the keyboard, I think it makes sense to remove the `self._keyboard` piece and use the module normally.